Application/src/main/java/practice/application/service/OrderService.java

```diff
@@ -23,45 +27,77 @@ public class OrderService {
 
     private final OrderRepository orderRepository;
     private final OrderItemsService orderItemsService;
+    private final MemberRepository memberRepository;
 
 
+
+    public List<OrderEntity> findEmail(String email){
+        List<OrderEntity> orderEntityList = orderRepository.findByEmail(email, OrderStatus.CANCELED).orElseThrow(() -> new NotFoundException("해당 이메일에 대한 주문은 없습니다"));
+
+        if(orderEntityList.isEmpty()){
+          throw new NotFoundException("해당 이메일에 대한 주문은 없습니다");
+        }
+
+        return orderEntityList;
+    }
+
     @Transactional
     public OrderEntity save(OrderCreateDTO orderCreateDTO) {
-
         List<OrdersItemEntity> orderItems = orderItemsService.createOrderItems(orderCreateDTO.getOrderItemsDTOS());
 
         OrderEntity orderEntity = new OrderEntity(orderCreateDTO.getEmail(), orderCreateDTO.getPostCode(), orderItems);
 
-        return orderRepository.save(orderEntity);
-    }
+        MemberEntity member = memberRepository.findByEmail(orderCreateDTO.getEmail())
+                .orElseThrow(() -> new NotFoundException("해당 회원을 찾을 수 없습니다"));
 
+        // 할인 적용 메서드 호출
+        applyMemberDiscount(orderEntity, member);
```

Comment:
회원 정보에 결제 금액도 변경된다는 메서드 이름이면 좀 더 보기 쉬울 것 같아요

```suggestion
recordDiscountedPrice(orderEntity, member);
```